File: 1-easy/00922-sort-array-by-parity-ii/solution.py
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:

        odd = 1
        even = 0
        nums_len = len(nums)

        while even < nums_len and odd < nums_len:
            if (nums[odd] & 1) != 1 or (nums[even] & 1) != 0:
                while (nums[odd] & 1) == 1:
                    odd += 2
                while (nums[even] & 1) == 0:
                    even += 2

                tmp = nums[even]
                nums[even] = nums[odd]
                nums[odd] = tmp

            even += 2
            odd += 2

        return nums

File: 1-easy/00922-sort-array-by-parity-ii/solution.rs
impl Solution {
    pub fn sort_array_by_parity_ii(mut nums: Vec<i32>) -> Vec<i32> {

        let mut odd = 1_usize;
        let mut even = 0_usize;
        let len = nums.len();

        while even < len && odd < len {
            if (nums[odd] & 1) != 1 || (nums[even] & 1) != 0 {
                while (nums[odd] & 1) == 1 {
                    odd += 2;
                }
                while (nums[even] & 1) == 0 {
                    even += 2;
                }

                let tmp = nums[even];
                nums[even] = nums[odd];
                nums[odd] = tmp;

            }

            even += 2;
            odd += 2;
        }

        nums
    }
}
